File: backend/src/main/java/com/poeflipfinder/backend/gateway/ExchangeChangeStreamPage.java
package com.poeflipfinder.backend.gateway;

import java.util.List;

/** One page of the Currency Exchange change-stream, crossing the ExchangeSourceGateway seam. */
public record ExchangeChangeStreamPage(List<ExchangeMarketEntry> entries, long nextChangeId, boolean atTip) {
}

File: app/lib/poe_flip_finder/exchange_change_stream_page.ex
defmodule PoeFlipFinder.ExchangeChangeStreamPage do
  @moduledoc "One page of the Currency Exchange change-stream, crossing the ExchangeSourceGateway seam."

  alias PoeFlipFinder.ExchangeMarketEntry

  @enforce_keys [:entries, :next_change_id, :at_tip]
  defstruct [:entries, :next_change_id, :at_tip]

  @type t :: %__MODULE__{
          entries: [ExchangeMarketEntry.t()],
          next_change_id: integer(),
          # Named `at_tip`, not `at_tip?` -- see docs/ELIXIR_CODE_STYLE.md;
          # the `?` suffix is reserved for functions, not struct fields.
          at_tip: boolean()
        }
end
